### DISCUSSION TITLE
How can I mark a table as incorrect?

### DISCUSSION THREAD

**Discussion by hgfrey** [primefaces/community#229] [PrimeFaces] [answered]
I have a table that must contain at least one row. If the user does not add a row, the table should be marked as incorrect. This is possible without any problems with a UIInput component (setValid), but not with a DataTable component. I have now placed a div element around the table and give it a red border if an error occurs, which I remove before checking again. This works, but requires very special programming. Is there no other way to include a table in the normal error process?

**Comment by jepsar** [comment]
I guess the answer here is: no.

**Reply by melloware** [reply, marked as answer]
@hgfrey they way you are doing it is the correct way as DataTable is not a UIInput component its a display component that knows nothing of validation.

**Reply by hgfrey** [reply]
Okay, thank you.
